### ISSUE TITLE
feature request: accordion widget

### ISSUE THREAD

**Issue by stephan-ansems** [beercss/beercss#19]
Hi,

I've been using beercss for a couple of days now and am impressed with the lightweightedness and ease of use. Chapeau! For a little project i'm working on i needed a accordion feature to group lots of settings more compactly on screen. 
I'm also using jQuery because i like it, but it should be possible to modify the code below to remove this dependency. I leave that to the guru's of beercss to do that. I'm no JS-buff...

So to add the accordion to my project I added something likt the following:
```html
<div class="accordion">
  <div><span class="title">Title of section one</span></div>
  <div>Content of section one</div>

  <div><span class="title">Title of section two</span></div>
  <div>Content of section two</div>

  <!-- etc. you get the picture, right? -->
</div>
```

I needed some additional CSS of course to style it nicely:
```css
/* GENERAL FORMATTING FOR ACCORDION ELEMENTS */
.accordion {
  background: #fff;
}

.accordion > * {
  padding: 0 10px;
  margin: 0;
}

/* FORMATTING FOR THE SECTION HEADERS */

/* common accordion header styling */
.accordion > *:nth-child(2n + 1) {
  background: #fff;
  border-bottom: 1px solid #ccc;
  display: flex;
  flex-grow: 1;
  flex-basis: auto;
  justify-content: space-between;
  height: 56px;
  align-items: center;
}

/* title component accordion header styling */
.accordion > *:nth-child(2n + 1) .title {
  margin-left: 8px;
  flex-grow: 1;
}

/* collapsed header icon */
.accordion > *:nth-child(2n + 1)::after {
  flex-basis: 24px;
  opacity: 0.5;
  content: url('data:image/svg+xml,%3Csvg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="%23000000"%3E%3Cpath d="M24 24H0V0h24v24z" fill="none" opacity=".87"/%3E%3Cpath d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"/%3E%3C/svg%3E');
}

/* expanded header styling */
.accordion > *.active:nth-child(2n + 1) {
  background-color: #8bd0bd;
}

/* expanded header icon */
.accordion > *.active:nth-child(2n + 1)::after {
  content: url('data:image/svg+xml,%3Csvg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="%23000000"%3E%3Cpath d="M0 0h24v24H0V0z" fill="none"/%3E%3Cpath d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14l-6-6z"/%3E%3C/svg%3E');
}

/* STYLING FOR CONTENT PANELS */

/* basic styling for all content panels */
.accordion > *:nth-child(2n) {
  background-color: #fff;
  display: flex;
  flex-flow: column nowrap;
  padding: 0 10px 20px 10px;
  position: relative;
  border-bottom: 1px solid #ccc;
  box-shadow: inset -2px -2px 4px 0px #e0e0e0;
}

/* close all except the first panel */
.accordion > *:nth-child(2n + 4) {
  display: none;
}
```

Finally to enable the accordion's functionality i needed to add some scripting, i used jQuery:
```javascript
  // Add the onClick behaviour...
  $('.accordion > *:nth-child(2n+1)').on('click', (e) => {
    // Find the section header closest the clicked element...
    var header = $(e.target).closest('.section-header'); 
    
    // Ignore the click if the header is already the active header...
    if (header.hasClass('active')) return false;

    // Remove the active style from all headers with style 'active' and 
    // roll up all their corresponding content panels...
    $(e.target)
      .closest('.accordion')
      .children('.active')
      .removeClass('active')
      .next()
      .slideUp();

    // Now style the clicked header as 'active' and slide the corresponding
    // content panel down...
    header.addClass('active')
       .next()
       .slideDown();
  });

  // Initial styling of the first section as active...
  $('.accordion > *:first').addClass('active');
  
  // Add a special class to all section headers for easier coding...
  $('.accordion > *').addClass('section-header');
```

**Comment by leonardorafael** [comment]
Hi my friends,

I'm working on some ideas here https://codepen.io/leo-bnu/pen/PoKQYON using `<details>` and `<summary>` for our expandable/accordion widgets. A solution with pure css without js. What do you think about it?

**Comment by leonardorafael** [comment]
Released in v1.2.7

**Comment by laurentpayot** [comment]
> using `<details>` and `<summary>` for our expansion/accordion widgets

Great idea :+1:
